axiom/nr_php_packages.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef nr_php_packages_HDR
+#define nr_php_packages_HDR
+
+#include "nr_log_event.h"
+#include "util_random.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+
+typedef struct _nr_php_package_t {
+  char* package_name;
+  char* package_version;
+} nr_php_package_t;
+
+typedef nr_hashmap_t nr_php_packages_t;
```

Comment:
I would suggest a different name for this hashmap- `nr_php_packages_t` does not indicate that it's actually a hashmap under the hood, and is very similar in name to the above struct `nr_php_package_t`. This can become confusing when trying to read and write code. 
Since we aren't changing anything about the underlying hashmap `nr_hashmap_t` implementation, do we need this typedef at all? Couldn't we just use the baseline `nr_hashmap_t` instead?

Reply:
The idea was if we did change the implementation then we are not going around changing the data type.  This is a common form of hiding the implementation details so one can make changes in the future w/o having to change any consumers of the library.

Reply:
We do this all over the place - `nr_log_event_t` and `nr_log_events_t`, etc.

Reply:
`nr_log_events_t` has a unique structure implementation though: https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/nr_log_events.c#L17 
It isn't just an alias for `nr_hashmap_t`

Reply:
I can't find any occurrence in the code where we use an alias for the `nr_hashmap_t` struct alone

Reply:
> The idea was if we did change the implementation then we are not going around changing the data type. This is a common form of hiding the implementation details so one can make changes in the future w/o having to change any consumers of the library.

I agree with the above. `nr_php_packages_t` with set of `nr_php_packages_*` functions hides the implementation nicely.

Reply:
> > We do this all over the place - `nr_log_event_t` and `nr_log_events_t`, etc.

I believe this was in response to 

> `nr_php_packages_t` [...] is very similar in name to the above struct `nr_php_package_t`. This can become confusing when trying to read and write code.

and has nothing to do with `nr_php_packages_t` being an alias of `nr_hashmap_t`.

Reply:
I think this is a better way to create the data structure for packages:
```
struct _nr_php_packages_t {
  nr_hashmap_t *data;
};

typedef nr_php_packages_t ) _nr_php_packages_t;
```

It is odd to see `nr_packages_t *` passed where the hashmap API expects `nr_hashmap_t *`.  This change would keep the isolation of how the php packages data type is implemented from consumers of the API and make the code more clear.

Reply:
Likewise `nr_php_package_t` probably could use a similar `struct` wrapper.